authorization-service/src/handlers/basic-authorizer.ts

```diff
@@ -0,0 +1,40 @@
+const { TEST_USER, TEST_PASSWORD } = process.env;
+
+const generatePolicy = (principalId, resource, effect = 'Allow') => ({
+  principalId,
+  policyDocument: {
+    Version: '2012-10-17',
+    Statement: [{
+      Action: 'execute-api:Invoke',
+      Effect: effect,
+      Resource: resource
+    }]
+  }
+});
+
+const basicAuthorizer = async (event, _, callback) =>  {
+  console.log(`Event data: ${JSON.stringify(event)}`);
+
+  try {
+    const { authorizationToken, methodArn } = event;
+    const encodedCredentials = authorizationToken.split(' ')[1];
+    const buffer = Buffer.from(encodedCredentials, 'base64');
+    const [ username, password ] = buffer.toString('utf-8').split(':');
+
+    console.log(`Username: ${username}, Password: ${password}`);
+
+    const effect = username === TEST_USER || password === TEST_PASSWORD ? 'Allow' : 'Deny';
```

Comment:
That is, if only the username is correct, authorization will still pass (I checked, it is so). Use && instead ||

Reply:
![image](https://user-images.githubusercontent.com/33957610/100719602-e51b0e80-33cd-11eb-97ff-249bd684ced0.png)
It is working even the authorization token is wrong. Please fix and reply.

Reply:
Updated

Reply:
It worked so as I wanted to check either login or password (it was for testing perspective). When you add smt at the end of the token the validation for login worked and it allowed you to proceed with this validation. Anyway, I updated and now it works as you expected

Reply:
Great. Now I can set you the 6 grade without any objections